### PR TITLE
Linter: USB-C trademark miss-spelling rule and content fix [PC-1014, PC-1015]

### PR DESCRIPTION
**Pull request by marqdevx**
## What This PR Changes
- Linter: New rule to highlight the USB-C typos
  - Fix USB-C trademark miss-spells

### CODE REVIEW

File: content/hardware/03.nano/carriers/nano-motor-carrier/tutorials/nano-matlab-wifi-led/nanoMatlabWiFiLED.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ The goals of this project are:
 
 - [Arduino Nano 33 IoT](https://store.arduino.cc/products/arduino-nano-33-iot)
 - [Arduino Nano Motor Carrier](https://store.arduino.cc/products/arduino-nano-motor-carrier)
-- [Micro USB Cable](https://store.arduino.cc/products/usb-cable-type-a-male-to-micro-type-b-male)
+- [Micro USB Cable](https://store.arduino.cc/products/USB-C®able-type-a-male-to-micro-type-b-male)
```
Comment (sebromero) [inline content/hardware/03.nano/carriers/nano-motor-carrier/tutorials/nano-matlab-wifi-led/nanoMatlabWiFiLED.md:34]: Shouldn't match URLs

File: content/hardware/03.nano/carriers/nano-motor-carrier/tutorials/nano-simulink-wifi-led/nanoSimulinkWiFiLED.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ The goals of this project are:
 
 - [Arduino Nano 33 IoT](https://store.arduino.cc/products/arduino-nano-33-iot)
 - [Arduino Nano Motor Carrier](https://store.arduino.cc/products/arduino-nano-motor-carrier)
-- [Micro USB Cable](https://store.arduino.cc/products/usb-cable-type-a-male-to-micro-type-b-male)
+- [Micro USB Cable](https://store.arduino.cc/products/USB-C®able-type-a-male-to-micro-type-b-male)
```
Comment (sebromero) [inline content/hardware/03.nano/carriers/nano-motor-carrier/tutorials/nano-simulink-wifi-led/nanoSimulinkWiFiLED.md:36]: Shouldn't match URLs

File: content/hardware/02.hero/boards/yun-rev2/tutorials/yun-first-config/content.md [hunk 3]
```diff
@@ -644,13 +644,13 @@ void loopSerialTerminal() {
 
   }
 
-  // copy from UART to USB-CDC
+  // copy from UART to USB-C®DC
```
Comment (sebromero) [inline content/hardware/02.hero/boards/yun-rev2/tutorials/yun-first-config/content.md:647]: Shouldn't match code blocks

File: content/hardware/02.hero/boards/yun-rev2/tutorials/yun-first-config/content.md [hunk 4]
```diff
@@ -580,9 +580,9 @@ void startSerialTerminal() {
 bool commandMode = false;
 void loopSerialTerminal() {
 
-  // copy from USB-CDC to UART
+  // copy from USB-C®DC to UART
 
-  int c = SERIAL_PORT_USBVIRTUAL.read();    // read from USB-CDC
+  int c = SERIAL_PORT_USBVIRTUAL.read();    // read from USB-C®DC
```
Comment (sebromero) [inline content/hardware/02.hero/boards/yun-rev2/tutorials/yun-first-config/content.md:585]: Shouldn't match code blocks

File: content/tutorials/projects/iot-air-quality-checker/iot-air-quality-checker.md [hunk 5]
```diff
@@ -15,7 +15,7 @@ source: "https://create.arduino.cc/projecthub/407590/iot-air-quality-checker-c8d
 - [Seeed Grove - Dust Sensor（PPD42NS）](https://www.seeedstudio.com/Grove-Dust-Sensor%EF%BC%88PPD42NS%EF%BC%89-p-1050.html)
 - [Arduino Seeed Seed - Grove - Oled Display 0.96"](https://store.arduino.cc/grove-oled-display-0-96)
 - [Seeed Grove - Air quality sensor v1.3](https://www.seeedstudio.com/Grove-Air-quality-sensor-v1.3-p-2439.html)
-- [Arduino USB cable type A male to micro type B male](https://store.arduino.cc/usb-cable-type-a-male-to-micro-type-b-male)
+- [Arduino USB cable type A male to micro type B male](https://store.arduino.cc/USB-C®able-type-a-male-to-micro-type-b-male)
```
Comment (sebromero) [inline content/tutorials/projects/iot-air-quality-checker/iot-air-quality-checker.md:18]: Same

File: scripts/validation/rules/rules-trademarks.yml [hunk 6]
```diff
@@ -1,15 +1,23 @@
...
 # Avoids false positives by excluding the word in conjunction with a dash (e.g. URLs) and as part of code)
 - regex: "(?<![-\\/])([lL]o[rR]a(?:WAN| Alliance)?)(?!®)(?![-\\/]|\\.?\\S)\\b"
   shouldMatch: false
   includeCodeBlocks: false
   format: markdown
   type: warning
   errorMessage: The LoRa® / LoRaWAN® trademark is not used correctly.
+
+# USB-C® / USB Type-C® / USB4® rule
+- regex: "(?:[uU][sS][bB]-[cC]|[uU][sS][bB] [tT]ype[ -cC][cC]|[uU][sS][bB][4](?!®)|usb-c)(?![-\\/]|\\.?\\S)"
```
Comment (sebromero) [inline scripts/validation/rules/rules-trademarks.yml:18]:
```suggestion
- regex: "(?<!http[s]?:\\/\\/\\S*)(?:USB-C|USB Type[ -]C|USB4)(?!®)\\b"
  regexModifiers: "gi"
```

**Review by sebromero — CHANGES_REQUESTED** [review]
See comments. You need to manually look through all changes when doing a search/replace action. The regex didn't work for me. Try with:

USB-C rocks!
USB-C® also!
http://www.usb-c.com
USB-Cable

**Comment by marqdevx** [comment]
Those errors where caused by the replace in vsc, my fault.
Fixed :)

**Review by sebromero — APPROVED** [review]
If you're confident there are no more false positives from the search & replaces operation we can merge it. I did not go through all the changes.

**Comment by marqdevx** [comment]
I need to check/fix the "fix the issues" task, should I do it in another PR?

**Pull request merged**